src/util/colorScale.js

```diff
@@ -202,58 +203,66 @@ function createContinuousScale(colorBy, t1nodes, t2nodes) {
       minMax = getMinMaxFromTree(t1nodes, t2nodes, colorBy);
   }
 
+  /* user-defined anchor points across the scale */
+  const anchorPoints = _validateContinuousAnchorPoints(providedScale);
+
   /* make the continuous scale */
   let domain, range;
-  switch (colorBy) {
-    case "num_date":
-      /* we want the colorScale to "focus" on the tip dates, and be spaced according to sampling */
-      let rootDate = getTraitFromNode(t1nodes[0], "num_date");
-      let vals = t1nodes.filter((n) => !n.hasChildren)
-        .map((n) => getTraitFromNode(n, "num_date"));
-      if (t2nodes) {
-        const treeTooRootDate = getTraitFromNode(t2nodes[0], "num_date");
-        if (treeTooRootDate < rootDate) rootDate = treeTooRootDate;
-        vals.concat(
-          t2nodes.filter((n) => !n.hasChildren)
-            .map((n) => getTraitFromNode(n, "num_date"))
-        );
-      }
-      vals = vals.sort();
-      domain = [rootDate];
-      const n = 10;
-      const spaceBetween = parseInt(vals.length / (n - 1), 10);
-      for (let i = 0; i < (n-1); i++) domain.push(vals[spaceBetween*i]);
-      domain.push(vals[vals.length-1]);
-      domain = [...new Set(domain)]; /* filter to unique values only */
-      range = colors[domain.length]; /* use the right number of colours */
-      break;
-    default:
-      range = colors[9];
-      domain = genericDomain.map((d) => minMax[0] + d * (minMax[1] - minMax[0]));
+  if (colorBy==="num_date") {
+    /* we want the colorScale to "focus" on the tip dates, and be spaced according to sampling */
+    let rootDate = getTraitFromNode(t1nodes[0], "num_date");
+    let vals = t1nodes.filter((n) => !n.hasChildren)
+      .map((n) => getTraitFromNode(n, "num_date"));
+    if (t2nodes) {
+      const treeTooRootDate = getTraitFromNode(t2nodes[0], "num_date");
+      if (treeTooRootDate < rootDate) rootDate = treeTooRootDate;
+      vals.concat(
+        t2nodes.filter((n) => !n.hasChildren)
+          .map((n) => getTraitFromNode(n, "num_date"))
+      );
+    }
+    vals = vals.sort();
+    domain = [rootDate];
+    const n = 10;
+    const spaceBetween = parseInt(vals.length / (n - 1), 10);
+    for (let i = 0; i < (n-1); i++) domain.push(vals[spaceBetween*i]);
+    domain.push(vals[vals.length-1]);
+    domain = [...new Set(domain)]; /* filter to unique values only */
+    range = colors[domain.length]; /* use the right number of colours */
+  } else if (anchorPoints) {
```

Comment:
The ordering here prevents one from defining a scale for `num_date`, but I can easily switch this if desired

Reply:
Updated & force pushed so that we can now define custom scales for sampling date